File: C++_Src/SeriesInfo.cpp
#include "pch.h"
#include "SeriesInfo.h"

SeriesInfo::SeriesInfo(){}

SeriesInfo::SeriesInfo(unsigned int id,string sd, string sdp, string siid, string sn, string st , string sl, string stn, string sbs, string sm)
{
	this->id = id;
	this->seriesDate = sd;
	this->seriesDescription = sdp;
	this->seriesInstanceId = siid;
	this->seriesNum = sn;
	this->seriesTime = st;
	this->sliceLocation = sl;
	this->sliceThickness = stn;
	this->spacingBetweenSlices = sbs;
	this->studyModality = sm;
}

SeriesInfo::~SeriesInfo(){}

SeriesInfo& SeriesInfo::operator=(const SeriesInfo &s)
{
	if (this != &s)
	{
		this->id = s.id;
		this->seriesDate = s.seriesDate;
		this->seriesDescription = s.seriesDescription;
		this->seriesInstanceId = s.seriesInstanceId;
		this->seriesNum = s.seriesNum;
		this->seriesTime = s.seriesTime;
		this->sliceLocation = s.sliceLocation;
		this->sliceThickness = s.sliceThickness;
		this->spacingBetweenSlices = s.spacingBetweenSlices;
		this->studyModality = s.studyModality;
	}
	return *this;
}

void SeriesInfo::SetId(unsigned int id)
{
	this->id = id;
}

unsigned int SeriesInfo::GetId()
{
	return this->id;
}

void SeriesInfo::SetSeriesNum(string sn)
{
	this->seriesNum = sn;
}

string SeriesInfo::GetSeriesNum()
{
	return this->seriesNum;
}

void SeriesInfo::SetSeriesInstanceId(string siid)
{
	this->seriesInstanceId = siid;
}

string SeriesInfo::GetSeriesInstanceId()
{
	return this->seriesInstanceId;
}

void SeriesInfo::SetStudyModality(string sm)
{
	this->studyModality = sm;
}

string SeriesInfo::GetStudyModality()
{
	return this->studyModality;
}

void SeriesInfo::SetSeriesDescription(string sd)
{
	this->seriesDescription = sd;
}

string SeriesInfo::GetSeriesDescription()
{
	return this->seriesDescription;
}

void SeriesInfo::SetSeriesDate(string sd)
{
	this->seriesDate = sd;
}

string SeriesInfo::GetSeriesDate()
{
	return this->seriesDate;
}

void SeriesInfo::SetSeriesTime(string st)
{
	this->seriesTime = st;
}

string SeriesInfo::GetSeriesTime()
{
	return this->seriesTime;
}

void SeriesInfo::SetSliceThickness(string st)
{
	this->sliceThickness = st;
}

string SeriesInfo::GetSliceThickness()
{
	return this->sliceThickness;
}

void SeriesInfo::SetSpacingBetweenSlices(string sbs)
{
	this->spacingBetweenSlices = sbs;
}

string SeriesInfo::GetSpacingBetweenSlices()
{
	return this->spacingBetweenSlices;
}

void SeriesInfo::SetSliceLocation(string sl)
{
	this->sliceLocation = sl;
}

string SeriesInfo::GetSliceLocation()
{
	return this->sliceLocation;
}

void SeriesInfo::SetStudyId(unsigned int id)
{
	this->studyId = id;
}

unsigned int SeriesInfo::GetStudyId()
{
	return this->studyId;
}

ostream & operator<<(ostream & os, const SeriesInfo & s)
{
	os <<"ID: "<<s.id<<"\n"
		<< "seriesDate:" << s.seriesDate << "\n"
		<< "seriesDescription:" << s.seriesDescription << "\n"
		<< "seriesInstanceId:" << s.seriesInstanceId << "\n"
		<< "seriesNum:" << s.seriesNum << "\n"
		<< "seriesTime:" << s.seriesTime << "\n"
		<< "sliceLocation:" << s.sliceLocation << "\n"
		<< "sliceThickness:" << s.sliceThickness << "\n"
		<<"studyModality:"<<s.studyModality<<"\n"
		<< "spacingBetweenSlices:" << s.spacingBetweenSlices << "\n";
	return os;
}

File: java_Src/src/entity/SeriesInfo.java
package entity;

public class SeriesInfo {
	public SeriesInfo() {}
	private int id;
	private String seriesNum;
	private String seriesInstanceId;
	private String studyModality;
	private String seriesDescription;
	private String seriesDate;
	private String seriesTime;
	private String sliceThickness;
	private String spacingBetweenSlices;
	private String sliceLocation;
	private int studyId;
	public void setId(int id)
	{
		this.id=id;
	}
	public int getId()
	{
		return this.id;
	}
	public void setStudyId(int sid)
	{
		this.studyId=sid;
	}
	public int getStudyId()
	{
		return this.studyId;
	}
	public String getSeriesNum() {
		return seriesNum;
	}
	public void setSeriesNum(String seriesNum) {
		this.seriesNum = seriesNum;
	}
	public String getSeriesInstanceId() {
		return seriesInstanceId;
	}
	public void setSeriesInstanceId(String seriesInstanceId) {
		this.seriesInstanceId = seriesInstanceId;
	}
	public String getStudyModality() {
		return studyModality;
	}
	public void setStudyModality(String studyModality) {
		this.studyModality = studyModality;
	}
	public String getSeriesDescription() {
		return seriesDescription;
	}
	public void setSeriesDescription(String seriesDescription) {
		this.seriesDescription = seriesDescription;
	}
	public String getSeriesDate() {
		return seriesDate;
	}
	public void setSeriesDate(String seriesDate) {
		this.seriesDate = seriesDate;
	}
	public String getSeriesTime() {
		return seriesTime;
	}
	public void setSeriesTime(String seriesTime) {
		this.seriesTime = seriesTime;
	}
	public String getSliceThickness() {
		return sliceThickness;
	}
	public void setSliceThickness(String sliceThickness) {
		this.sliceThickness = sliceThickness;
	}
	public String getSpacingBetweenSlices() {
		return spacingBetweenSlices;
	}
	public void setSpacingBetweenSlices(String spacingBetweenSlices) {
		this.spacingBetweenSlices = spacingBetweenSlices;
	}
	public String getSliceLocation() {
		return sliceLocation;
	}
	public void setSliceLocation(String sliceLocation) {
		this.sliceLocation = sliceLocation;
	}
	
}
